arduino/libraries/librariesmanager/install.go

```diff
@@ -259,22 +260,49 @@ func parseGitURL(gitURL string) (string, error) {
 	return res, nil
 }
 
-// validateLibrary verifies the dir contains a valid library, meaning it has both
-// an header <name>.h, either in src or root folder, and a library.properties file
-func validateLibrary(name string, dir *paths.Path) error {
-	// Verify library contains library header.
-	// Checks also root folder for legacy reasons.
-	// For more info see:
-	// https://arduino.github.io/arduino-cli/latest/library-specification/#source-code
-	libraryHeader := name + ".h"
-	if !dir.Join("src", libraryHeader).Exist() && !dir.Join(libraryHeader).Exist() {
-		return fmt.Errorf(tr(`library is not valid: missing header file "%s"`), libraryHeader)
+// validateLibrary verifies the dir contains a valid library, meaning it has either
+// library.properties file and an header in src/ or an header in its root folder.
+// Returns nil if dir contains a valid library, error on all other cases.
+func validateLibrary(dir *paths.Path) error {
+	if dir.NotExist() {
+		return fmt.Errorf(tr("directory doesn't exist: %s", dir))
+	}
+	re := regexp.MustCompile(`.hp*$`)
```

Comment:
I don't think this is appropriate. It will match on invalid extensions like `.hpppppp` and it will miss the other allowed header `.hh`:
https://github.com/arduino/arduino-cli/blob/71ce1a46009ab15296e9a877280505b62124fcaa/arduino/libraries/libraries.go#L236

There should be a single list of valid header extensions that is used everywhere. Something like this:
https://github.com/arduino/arduino-cli/blob/71ce1a46009ab15296e9a877280505b62124fcaa/arduino/globals/globals.go#L46-L51
Arduino Lint could make use of it too if it was exported. I had to duplicate the list in Arduino Lint since Arduino CLI didn't provide it via the API.

Reply:
Good point, didn't think about it. 🤔